src/QMCHamiltonians/QMCHamiltonian.cpp

```diff
@@ -537,6 +537,49 @@ void QMCHamiltonian::auxHevaluate(ParticleSet& P, Walker_t& ThisWalker)
     auxH[i]->setParticlePropertyList(P.PropertyList,myIndex);
   }
 }
+///Evaluate properties only.
```

Comment:
@jtkrogel Do we have Hamiltonians or estimators contributing to both properties and collectables? if this is true, @rcclay calling auxHevaluateProperties is not safe.
@rcclay I feel that the code duplication can be avoided by introducing an extra optional argument for picking the three mode, both(default), properties, collectables.

Reply:
I thought all propertylist contributing QMCHamiltonianBase objects were explicitly required to not be collectable.  See e.g. QMCHamiltonianBase::addValue.

Reply:
@ye-luo Something looks odd with setParticlePropertyList in this regard.  In the regular code, setParticlePropertList is called for all auxH, but I don't see any guard against collectables.  In the case of collectables, it looks like myIndex remains set to -1; is there a sneaky index trick to have all collectables write to the same (safe) place by using index of -1?

Reply:
In any case this does not look any worse than what is there already...

Reply:
OK, now I see (remembering).  Collectables always need to define the virtual setParticlePropertyList function to be empty.  Will have to check if some do not do this...

Reply:
... only true if using hdf5 exclusively.

Reply:
Some collectable estimators use either one, e.g. MomentumEstimator, while others, e.g. SkAllEstimator just use the propertylist.

Reply:
Why is the call to auxHevaluateProperties not safe in this case?

Reply:
I really feel the handling of data collection needs to be looked at.
PropertyList takes care of the scalar.dat for small dataset.
Collectables takes care of h5 for large dataset.

Reply:
It is a bit odd.  I expect writing of collectables to scalar.dat has been done for the relative convenience/accessibility/robustness of post processing ascii files rather than hdf5.  I'm OK with it so long as both paths are provided correctly.

Reply:
@jtkrogel auxHevaluateProperties checks the QMCHamiltonianBase::COLLECTABLE flag. If true, skip the evaluation.

Reply:
OK.  In that case there is something to address.

Reply:
@jtkrogel You are right. How to IO (text or binary) is a separated issue. What I need to be clear is what should go to propertylist and what should be in collectables. propertylist should be minimized since it is carried by the walker.

Reply:
@ye-luo Good point.  At the very least all collectables should support hdf5 and, IMO, use it by default.

Reply:
@rcclay once you merge all these routines, I will approve this PR.

Reply:
@ye-luo How does it look now?  At the moment, I prefer to keep the auxHevaluate() used by the rest of the code, and my overloaded auxHevaluate(do_properties,do_collectables) separate.  I can merge them even more if need be.

Reply:
@rcclay Good for now.